src/wallet/wallet.cpp

```diff
@@ -2021,43 +2030,43 @@ int CWallet::ScanForWalletTransactions(CBlockIndex* pindexStart, bool fUpdate)
 
             CBlock block;
             ReadBlockFromDisk(block, pindex);
-            std::vector<const CTransactionBase*> vTxBase;
-            block.GetTxAndCertsVector(vTxBase);
   
-            for (const CTransactionBase* obj: vTxBase)
+            for(const CTransaction& tx: block.vtx)
             {
-                int bwtMaxDepth = -1;
-                bool areBwtStripped = false;
+                for(unsigned int pos = 0; pos < tx.GetVscCcOut().size(); ++pos)
+                    SyncSidechain(tx.GetScIdFromScCcOut(pos), CMinimalSidechain(CScCertificate::EPOCH_NULL, uint256()));
 
-                if (obj->IsCertificate())
-                {
-                    int nHeight = pindex->nHeight;
-                    const CScCertificate& cert = dynamic_cast<const CScCertificate&>(*obj);
-                    CSidechain sidechain;
-                    assert(pcoinsTip->GetSidechain(cert.GetScId(), sidechain));
-                    int currentEpoch = sidechain.EpochFor(nHeight);
-                    bwtMaxDepth = sidechain.StartHeightForEpoch(currentEpoch+1) +
-                        sidechain.SafeguardMargin() - nHeight;
-
-                    // Certs can have their bwts cleared for different reasons
-                    // ( being spent, voided due to sidechain ceasing, voided since cert has low quality)
-                    // which we are not able to reconstruct here.
-                    // Currently we just void all the bwts here
-                    if (pcoinsTip->IsBwtStripped(cert.GetHash()) )
-                    {
-                        LogPrint("cert", "%s():%d - cert[%s] has bwt stripped\n",
-                            __func__, __LINE__, cert.GetHash().ToString());
-                        areBwtStripped = true;
-                    }
-                }
+                if (AddToWalletIfInvolvingMe(tx, &block, -1, fUpdate))
```

Comment:
Consider syncing only if AddToWalletIfInvolvingMe() returns true

Reply:
Unlike SyncVoidingCert, which requires cert to be added to wallet map before being voided, syncSidechain is independent.